lib/config/screen.ex

```diff
@@ -122,12 +122,12 @@ defmodule ScreensConfig.Screen do
     |> String.ends_with?(@v2_app_id_suffix)
   end
 
-  for vendor <- ~w[gds mercury solari mimo c3ms outfront]a do
-    vendor_string = Atom.to_string(vendor)
+  defp value_from_json("vendor", vendor_string, _app_id) when vendor_string in ["n/a", ""] do
+    :n_a
+  end
 
-    defp value_from_json("vendor", unquote(vendor_string), _app_id) do
-      unquote(vendor)
-    end
+  defp value_from_json("vendor", vendor_string, _app_id) do
+    String.to_existing_atom(vendor_string)
```

Comment:
This would technically allow us to enter any value for a vendor that corresponds to an existing atom, which not only is a big set of values, it could change between deploys or even at runtime. I think I like the previous approach better where the vendors are strictly enumerated (though that does mean the list has to be duplicated in the typespec) — we could handle unknown values by having a clause like

```elixir
defp value_from_json("vendor", other, _app_id), do: raise "unknown vendor #{other}"
```

Reply:
Very good point. Resolve the issue I raised in Slack as well.